web/job_runs_controller_test.go

```diff
@@ -20,8 +20,16 @@ type JobRun struct {
 	ID string `json:"id"`
 }
 
+func BenchmarkJobRunsController_Index(b *testing.B) {
+	testJobRunsControllerIndex(b)
+}
+
 func TestJobRunsController_Index(t *testing.T) {
 	t.Parallel()
+	testJobRunsControllerIndex(t)
+}
+
+func testJobRunsControllerIndex(t assert.TestingT) {
```

Comment:
I think we should separate out the applications setup and saving of records from outside the benchmark, as they are definitely the slowest part and only really need to be run once. Similarly, let's remove the assertions at the end of the benchmark, expect a check for successful response code.

Reply:
Also, probably call `b.ResetTimer()` just before running the benchmarking loop.

Reply:
Sure, I'll get that re-jigged. It just felt wrong duplicating most of the methods and leaving them mostly the same. Especially with the helper methods as well, since a lot of them are fixed to using Testing.T, I'd have to copy and paste and change them to use Testing.B.

I started going about it this way, but there was an awful lot of duplication. I'll relook at this and see if there's any better way to go about it.

Regarding the speed of the app set-up, I didn't deem it as much of an issue since the benchmarks were recording averages of 0.5ns.